packages/suite-desktop/src-electron/processes/BaseProcess.ts

```diff
@@ -0,0 +1,86 @@
+import path from 'path';
+
+import { getPlatformExt, killProcess, runProcess } from '../utils/process';
+import { RESOURCES } from '../constants';
+
+export type Status = {
+    service: boolean;
+    process: boolean;
+};
+
+class BaseProcess {
+    processName = '';
+    resourceName = '';
+    runBySuite = false;
+
+    /**
+     * Returns the status of the service/process
+     * - service: The service is working
+     * - process: The process is running
+     */
+    async status() {
+        return {
+            service: false,
+            process: false,
+        };
+    }
+
+    /**
+     * Start the bundled process
+     * @param force Force the launch (will also terminate any other processes with that name)
+     * @param params Command line parameters for the process
+     */
+    async start(force = false, params: string[] = []) {
+        const { process, service } = await this.status();
+
+        // Service is running, nothing to do
+        if (!force && service) {
+            return;
+        }
+
+        // If the process is running but the service isn't
+        if (process || force) {
+            // Stop the process
+            await this.stop(!force);
+        }
+
+        const extension = getPlatformExt();
+        const processPath = path.join(RESOURCES, this.resourceName, this.processName);
```

Comment:
```suggestion
        const processPath = path.join(RESOURCES, 'bin', this.resourceName, this.processName);
```

Reply:
make sure that this `bin` is not platform specific

Reply:
> make sure that this `bin` is not platform specific (ex: windows)

No problem here, `bin` is our folder, not system on.